### PR TITLE
feat(qrm): use async worker to set extra cg mem limit to avoid thread stucking

### PR DESCRIPTION
**Pull request by csfldf**
feat(qrm): use async worker to set extra cg mem limit to avoid thread stucking

### CODE REVIEW

**Review by cheney-lin — DISMISSED** [review]

**Review by cheney-lin — APPROVED** [review]

File: pkg/util/cgroup/manager/cgroup.go [hunk 1]
```diff
@@ -292,7 +292,54 @@ func DropCacheWithTimeoutWithRelativePath(timeoutSecs int, absCgroupPath string,
...
+	}
+
+	return nil
+}
+
+func SetExtraCGMemLimitWithTimeoutAndRelCGPath(ctx context.Context, relCgroupPath string, timeoutSecs int, nbytes int64) error {
+	memoryAbsCGPath := common.GetAbsCgroupPath(common.CgroupSubsysMemory, relCgroupPath)
+
+	err := SetExtraCGMemLimitWithTimeoutAndAbsCGPath(timeoutSecs, memoryAbsCGPath, nbytes)
+	_ = asyncworker.EmitAsyncedMetrics(ctx, metrics.ConvertMapToTags(map[string]string{
```
Comment (zzzzhhb) [inline pkg/util/cgroup/manager/cgroup.go:310]: should this metric logic be done in asyncworker's scope? what if this method wasn't called by asyncworker?

File: pkg/util/cgroup/manager/cgroup.go [hunk 2]
```diff
@@ -292,7 +292,54 @@ func DropCacheWithTimeoutWithRelativePath(timeoutSecs int, absCgroupPath string,
...
+		//cgv2
+		interfacePath = filepath.Join(absCgroupPath, "memory.max")
+	} else {
+		//cgv1
+		interfacePath = filepath.Join(absCgroupPath, "memory.limit_in_bytes")
+	}
+
+	cmd := fmt.Sprintf("timeout %d echo %d > %s", timeoutSecs, nbytes, interfacePath)
+
+	_, err := exec.Command("bash", "-c", cmd).Output()
```
Comment (zzzzhhb) [inline pkg/util/cgroup/manager/cgroup.go:339]: just curious, why don't we just use io library instead of a bash command?

Comment (csfldf) [reply pkg/util/cgroup/manager/cgroup.go:339]: because we want to reuse TIMEOUT(1)  user commands to trigger timeout to prevent stucking

Comment (zzzzhhb) [reply pkg/util/cgroup/manager/cgroup.go:339]: get

File: pkg/util/cgroup/manager/cgroup.go (hunk 1 above)
Comment (csfldf) [reply pkg/util/cgroup/manager/cgroup.go:310]: [TODO]
1. use async worker framework to record succees information for works
2. pass and use normal emitter to record specic information for works

**Review by zzzzhhb — APPROVED** [review]

**Pull request merged**